source/extensions/retry/host/omit_canary_hosts/config.h

```diff
@@ -0,0 +1,29 @@
+#include "envoy/upstream/retry.h"
+
+#include "extensions/retry/host/omit_canary_hosts/omit_canary_hosts.h"
+#include "extensions/retry/host/well_known_names.h"
+
+namespace Envoy {
+namespace Extensions {
+namespace Retry {
+namespace Host {
+
+class OmitCanaryHostsRetryPredicateFactory : public Upstream::RetryHostPredicateFactory {
+  
+public:
+  Upstream::RetryHostPredicateSharedPtr createHostPredicate(const Protobuf::Message&,
+                                                            uint32_t) override {
+    return std::make_shared<OmitCanaryHostsRetryPredicate>();
+  }
+
+  std::string name() override { return RetryHostPredicateValues::get().OmitCanaryHostsPredicate; }
+
+  ProtobufTypes::MessagePtr createEmptyConfigProto() override {
+    return ProtobufTypes::MessagePtr{new Envoy::ProtobufWkt::Empty()};
```

Comment:
Use `std::make_unique` instead of explicit `new`

Reply:
Sure! I got started on this PR by copying the sources of the `previous_hosts` extensions, can I make this change there also?

Reply:
Sure, I wouldn't mind the cleanup